### PR TITLE
refactor(resharding) - Replacing usages of ShardLayout::v1 with multi_shard

### PR DESCRIPTION
**Pull request by wacban**
I'm replacing the usages of the v1 layout with v2 in tests to get as much coverage as possible for the new shard layout. No bugs found this time. I haven't managed to remove all of those but making progress. 

Some effort was required in the congestion control test because those used the clients as the parties in the test which got kind of messy. Instead I added dedicated accounts to perform the business logic. 

Please note that I'm intentionally keeping the new code under version agnostic multi_shard rather than using v2() -> this is so that in the future if more versions are added we can just change it in one place.

### CODE REVIEW

File: chain/client/src/test_utils/test_env.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 use crate::{Client, DistributeStateWitnessRequest};
+use core::panic;
```
Comment (marcelo-gonzalez) [inline chain/client/src/test_utils/test_env.rs:2]: did you mean to add this?

**Review by marcelo-gonzalez — APPROVED** [review]

File: chain/client/src/test_utils/test_env.rs [hunk 2]
```diff
@@ -139,6 +139,26 @@ impl TestEnv {
         self.process_block(id, block.unwrap(), Provenance::PRODUCED);
     }
 
+    // Produces block by the client that is the block producer for the given height.
+    pub fn produce_block_simple(&mut self, height: BlockHeight) {
+        let client = &self.clients[0];
+        let tip = client.chain.head().unwrap();
+        let epoch_id = tip.epoch_id;
```
Comment (Trisfald) [inline chain/client/src/test_utils/test_env.rs:146]: It probably doesn't matter in this case, but I thought `epoch_id` had to be retrieved from `height` not from `tip`

File: integration-tests/src/tests/client/flat_storage.rs [hunk 3]
```diff
@@ -455,42 +456,44 @@ fn test_flat_storage_iter() {
...
+            assert_eq!(
+                TrieKey::Account { account_id: "test0".parse().unwrap() }.to_vec(),
+                items[0].as_ref().unwrap().0.to_vec()
+            );
+        }
+        if shard_id == s2 {
+            // Test1 account was not created yet - so no entries.
+            // (with newer protocol: +1 for BandwidthSchedulerState)
+            let expected = 0 + protocol_version_modifier;
+            assert_eq!(expected, items.len());
```
Comment (Trisfald) [inline integration-tests/src/tests/client/flat_storage.rs:496]: Not related to this change: we could add an `assert_matches` on `TrieKey::BandwidthSchedulerState`

**Review by Trisfald — APPROVED** [review]
🚀

File: chain/client/src/test_utils/test_env.rs (hunk 2 above)
Comment (wacban) [reply chain/client/src/test_utils/test_env.rs:146]: oh my good point, let me look into this

**Pull request merged**